src/libraries/Microsoft.Extensions.Primitives/src/StringSegment.cs

```diff
@@ -428,7 +428,7 @@ public int IndexOf(char c, int start, int count)
         {
             int offset = Offset + start;
 
-            if (!HasValue || start < 0 || (uint)offset > (uint)Buffer.Length)
+            if (start < 0 || (uint)offset > (uint)Buffer.Length)
```

Comment:
>Buffer [](start = 50, length = 6)

It is possible the Buffer can be null when HasValue is false. You need to do the `(uint)offset > (uint)Buffer.Length` check only when HasValue is true.

Reply:
@hrrrrustic could you please address this last feedback so we can move forward merging this change? thanks for your help with that.

Reply:
@hrrrrustic did you have a chance to look at the remaining feedback? we are trying to close the long-aged PR's. Please let's know or you need help someone address the remaining feedback. Thanks!

Reply:
@hrrrrustic could you please respond? Thanks.

Reply:
@tarekgh I'm sorry, a little bit busy at university. I hope I can fix it this week (probably Thursday/Friday). If necessary, you can convert this PR to draft or close it and I'll reopen it later

Reply:
thanks @hrrrrustic. I'll close the PR for now to not put any pressure on you. You can reopen at any time you are ready. thanks again for your help all the way.